lib/homepage.dart

```diff
@@ -373,56 +417,50 @@ class _MyHomePageState extends State<MyHomePage> {
 
   @override
   Widget build(BuildContext context) {
-    // This method is rerun every time setState is called, for instance as done
-    // by the _incrementCounter method above.
-    //
-    // The Flutter framework has been optimized to make rerunning build methods
-    // fast, so that you can just rebuild anything that needs updating rather
-    // than having to individually change instances of widgets.
-    if (highlightsLoaded &&
-        webinarsLoaded &&
-        eventsLoaded &&
-        offersLoaded &&
-        announcementsLoaded)
-      return RefreshIndicator(
-          onRefresh: _refreshData,
-          child: WillPopScope(
-            onWillPop: () async => false,
-            child: Scaffold(
-              appBar: AppBar(
-                // Here we take the value from the MyHomePage object that was created by
-                // the App.build method, and use it to set our appbar title.
-                title: Text("Home"),
-                centerTitle: true,
-                backgroundColor: Utils.header,
-                flexibleSpace: Container(
-                  decoration: BoxDecoration(
-                    gradient: LinearGradient(
-                      colors: [
-                        Utils.secondaryColor,
-                        Utils.primaryColor,
-                      ],
-                      begin: Alignment.topRight,
-                      end: Alignment.bottomLeft,
+    return FutureBuilder<void>(
```

Comment:
`FutureBuilder` here makes more sense than doing the API calls synchronously then checking when the data is loaded.